src/modules/commander/Commander.cpp

```diff
@@ -2226,56 +2226,42 @@ Commander::run()
 		}
 
 		/* Check for failure detector status */
-		const bool failure_detector_updated = _failure_detector.update(status);
-
-		if (failure_detector_updated) {
-
-			const uint8_t failure_status = _failure_detector.getStatus();
-
-			if (failure_status != status.failure_detector_status) {
-				status.failure_detector_status = failure_status;
-				_status_changed = true;
-			}
-		}
-
-		if (armed.armed &&
-		    failure_detector_updated) {
-
-			if (_failure_detector.isFailure()) {
+		if (_failure_detector.update(status)) {
+			status.failure_detector_status = _failure_detector.getStatus();
+			_status_changed = true;
 
-				const hrt_abstime time_at_arm = armed.armed_time_ms * 1000;
+			if (armed.armed) {
+				if (status.failure_detector_status & vehicle_status_s::FAILURE_ARM_ESC) {
+					const hrt_abstime time_at_arm = armed.armed_time_ms * 1000;
 
-				if (hrt_elapsed_time(&time_at_arm) < 500_ms) {
 					// 500ms is the PWM spoolup time. Within this timeframe controllers are not affecting actuator_outputs
-
-					if (status.failure_detector_status & vehicle_status_s::FAILURE_ARM_ESC) {
+					if (hrt_elapsed_time(&time_at_arm) < 500_ms) {
 						arm_disarm(false, true, &mavlink_log_pub, arm_disarm_reason_t::FAILURE_DETECTOR);
 						_status_changed = true;
```

Comment:
`_status_changed` is already set to `true` above

Reply:
Let me check

Reply:
I corrected it in "Commander: remove duplicate flag changes". Thanks for the catch :+1: